src/main/java/org/cactoos/scalar/Folded.java

```diff
@@ -23,108 +23,68 @@
  */
 package org.cactoos.scalar;
 
-import java.util.Iterator;
-import java.util.NoSuchElementException;
 import org.cactoos.BiFunc;
 import org.cactoos.Scalar;
-import org.cactoos.iterable.Mapped;
+import org.cactoos.iterable.IterableOf;
 
 /**
- * Folds iterable via BiFunc.
+ * Iterable, which elements are "folded" through the func.
  *
- * <pre>
- * new Folded<>(
- *     (first, last) -> first + last,
- *     new IterableOf<>(() -> 1L, () -> 2L, () -> 3L, () -> 4L)
- * ).value() // returns 10L
- * </pre>
- *
- * <p>Here is how you can use it to
- * find one of items according to the specified {@link BiFunc}:</p>
- *
- * <pre>
- *  final String apple = new Folded&lt;&gt;(
- *          (first, last) -&gt; first,
- *          new IterableOf&lt;Scalar&lt;String&gt;&gt;(
- *              () -&gt; "Apple",
- *              () -&gt; "Banana",
- *              () -&gt; "Orange"
- *          )
- *      ).value();
- *  final String orange = new Folded&lt;&gt;(
- *          (first, last) -&gt; last,
- *          new IterableOf&lt;Scalar&lt;String&gt;&gt;(
- *              () -&gt; "Apple",
- *              () -&gt; "Banana",
- *              () -&gt; "Orange"
- *          )
- *      ).value();
- * </pre>
- *
- * <p>There is no thread-safety guarantee.
- *
- * <p>This class implements {@link Scalar}, which throws a checked
- * {@link Exception}. This may not be convenient in many cases. To make
- * it more convenient and get rid of the checked exception you can
- * use {@link UncheckedScalar} or {@link IoCheckedScalar} decorators.</p>
- *
- * @author Alexander Dyadyushenko (gookven@gmail.com)
- * @author Eduard Balovnev (bedward70@mail.ru)
+ * @author Yegor Bugayenko (yegor256@gmail.com)
  * @version $Id$
- * @param <T> Scalar type
- * @since 0.29
+ * @param <T> Type of element
+ * @param <X> Type of input and output
+ * @since 0.30
```

Comment:
@paulodamaso I'm afraid if changing the `@since` tag is a good idea. Classes and their functionality existed before and you just changed their names/content. I think it would be better to keep the versions of the functionality: Folded -> since  0.9, Reduced -> since 0.29. WDYT?

Reply:
@proshin-roman I'm thinking about naming, authoring and functionality/content: the file `Folded.java`, whose author is Yegor and which represents an "iterable which elements are 'folded' througn the function" appeared just in version 0.30, not before; the same with `Reduced.java`: we have not this name-author-content configuration (`Reduced.java`, which "Reduces iterable by BiFunc", by Alexander and Eduard) in version 0.29. Keeping the old version does not seems right to me because we would not respect these authoring information against name, version and content of each class.

Reply:
@paulodamaso To be honest, I don't think that authoring is the most important info in the code :) In any case, the code is changing constantly by many developers and at some point the class contains nothing written by the mentioned author. What is more important IMO is a relation of name, version, and content. Let's say, somebody uses the library of version 0.5 (for example). Then he can look at Folded class and find the version when this class has appeared the first time (0.29). And then he can just download this specific 0.29 version and then this file MUST be in the code. The problem that the class in version 0.29 has absolutely different functionality comparing to the same file in version 0.30. 

So... yes, I agree that we should specify the current version (0.30) - just because it's safer.